samples/common/mcumgr_bt_ota_dfu/CMakeLists.txt

```diff
@@ -0,0 +1,41 @@
+#
+# Copyright (c) 2023 Nordic Semiconductor ASA
+#
+# SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+#
+
+if (CONFIG_NCS_SAMPLE_MCUMGR_BT_OTA_DFU_VALIDATION)
+
+  if (CONFIG_NCS_SAMPLE_MCUMGR_BT_OTA_DFU_SPEEDUP AND CONFIG_BT_CTLR)
+    if ((NOT (CONFIG_BT_BUF_ACL_TX_SIZE EQUAL 251)) OR
+        (NOT (CONFIG_BT_BUF_ACL_RX_SIZE EQUAL 502)) OR
+        (NOT (CONFIG_BT_CTLR_DATA_LENGTH_MAX EQUAL 251))
+      )
+      message(WARNING "MCUmgr Bluetooth OTA DFU: suboptimal configuration "
+        "of Bluetooth Controller buffers.")
+    endif()
+  endif()
+
+  if (CONFIG_NCS_SAMPLE_MCUMGR_BT_OTA_DFU_SPEEDUP AND CONFIG_BT_HCI_HOST)
+    if ((NOT (CONFIG_BT_BUF_ACL_TX_SIZE EQUAL 251)) OR
+        (NOT (CONFIG_BT_BUF_ACL_RX_SIZE EQUAL 502)) OR
+        (NOT (CONFIG_BT_L2CAP_TX_MTU EQUAL 498))
+      )
+      message(WARNING "MCUmgr Bluetooth OTA DFU: suboptimal configuration "
+        "of Bluetooth Host buffers.")
```

Comment:
exactly why are those values suboptimal ?

and how does a user know what the optimal values will be.
Of course I can see that optimal values are those stated in the if checks above, but I don't expect a user / developer to go read the CMake files in order to understand what the optimal values are.

If we warn the user that something is suboptimal, then we should also provide the user with sufficient information so that he knows how to fix / silence the warning.

This could be by providing references to adequate documentation or maybe Kconfig symbols of importance.

I assume such documentation already exists somewhere, it's just that I don't knwo where to find it 😉

Reply:
Good point. I named the suboptimal configurations in each warning and pointed the user to the file with the optimal ones